packages/hyperion-autologging/src/ALSurface.ts

```diff
@@ -26,33 +26,34 @@ export type ALChannelSurfaceEventData = ALMetadataEvent & ALFlowletEvent & Reado
   surface: string;
   element: Element | null | undefined;
   isProxy: boolean;
-  capability: ALSurfaceCapability;
+  capability: ALSurfaceCapability | null | undefined;
 }>;
 
-export enum ALSurfaceCapability {
-  TrackInteraction = 1 << 0, // Mark all interaction events with this surface
-  TrackMutation = 1 << 1, // report mount/unmount of this surface
-  // TrackVisibility = 1 << 2, // track when mounted surface's visibility changes
-  // TrackBoundingRect = 1 << 3, // Report the size of the bounding rect of the surface on the screen.
+export interface ALSurfaceCapability {
+  /**
+   * By default, in addition to reporting mount/unmount of a surface, all
+   * interactions are also marked with a the name of their surface, unless
+   * the following flag is set.
+   */
+  nonInteractive?: boolean;
+
+  /**
+   * When set, will track when the provided ratio [0,1] of the surface becomes visible
+   */
+  trackVisibilityThreshold?: number;
 }
 
-const AllSurfaceCapabilityValues = [
-  ALSurfaceCapability.TrackInteraction,
-  ALSurfaceCapability.TrackMutation,
-];
-const AllSurfaceCapabilityValuesString = AllSurfaceCapabilityValues.map(c => ALSurfaceCapability[c]).join(',');
 function surfaceCapabilityToString(capability?: ALSurfaceCapability): string {
   if (!capability) {
-    return AllSurfaceCapabilityValuesString;
+    return '';
```

Comment:
Would default just be interactive?  Is it worth populating if it's in the default state?

Reply:
the reason I renamed the values is that capability only captures the features specified, so that everything that is not specified is default value as well. That's why I chose `nonInteractive` as field name as opposed to `interactive`. 

I'm not sure if we can have a good default value for all new values going forward. That's why it is left empty here. The value here would match the props of the surface.